platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsAccessFromKotlinDslIntegrationTest.groovy

```diff
@@ -121,4 +121,35 @@ class IsolatedProjectsAccessFromKotlinDslIntegrationTest extends AbstractIsolate
             problem("Build file 'a/build.gradle.kts': Project ':a' cannot access 'Project.buildDir' functionality on subprojects of project ':'", 2)
         }
     }
+
+    def "build script can query basic details of projects in a function called from allprojects block"() {
+        createDirs("a", "b")
+        settingsFile << """
+            rootProject.name = "root"
+            include("a", "b")
+        """
+        buildKotlinFile << """
+            fun printInfo(p: Project) {
+                println("project name = " + p.name)
+            }
+
+            allprojects {
```

Comment:
❓ What value are we getting from this test?

Reply:
As we discussed, this test mirrors the similar one for Groovy DSL and locks in the behavior that, in Kotlin DSL, there is no violation.
https://github.com/gradle/gradle/blob/2d1a6c0544d7287160a6bfdba82e7948f30583f0/platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsAccessFromGroovyDslIntegrationTest.groovy#L669

Perhaps the test needs a better name. Do you have ideas?

Reply:
It still feel a bit wasteful in compute time. I mean, this would/should be covered elsewhere by tests that use `allprojects` when running with `isolatedProjectsIntegTest`...

Reply:
I would have thought the same about the Groovy member resolution that was not caught by tests before 🙃

Reply:
I mean, we're just now adding Isolated Projects test coverage to `:core` (#29202) and we do see some Groovy related errors there. No harm in having some explicit test coverage I guess.